### ISSUE TITLE
False negatives for implicit string concatenation

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [PyCQA/pylint#6441]
### Current problem

The following code:
```python
print("Lorem ipsum " "dolor sit amet")
```
Could be rewritten:
```python
print("Lorem ipsum dolor sit amet")
```

But 
```python
print(
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit,"
    " sed do eiusmod tempor incididunt ut labore et dolore "
    "magna aliqua. Ut enim ad minim veniam, quis nostrud "
    "exercitation ullamco laboris nisi ut aliquip ex ea "
)
```
Should not, because the implicit string concatenation is not useless as removing it would make the line go over the max line length.

### Desired solution

Raise ``implicit-str-concat`` in this situation.

### Additional context

https://github.com/PyCQA/astroid/pull/1525#discussion_r856578221

**Comment by Pierre-Sassoulas** [comment]
A decision was made in fcc01516ae to not raise ``implicit-str-concat`` on function call, I don't understand the rationale.

**Comment by cdce8p** [comment]
> A decision was made in [fcc0151](https://github.com/PyCQA/pylint/commit/fcc01516ae176ad3fdedc4497328105f3314e376) to not raise `implicit-str-concat` on function call, I don't understand the rationale.

Saw this comment about it: https://github.com/PyCQA/pylint/pull/1655#issuecomment-328760196

**Comment by Pierre-Sassoulas** [comment]
We have [the option ``check-str-concat-over-line-jump`` to handle multi line implicit str concat though](https://github.com/PyCQA/pylint/blob/main/pylintrc#L556), the example in this comment would be treated well. Also I think I have a good example of a call being faulty if you consider ``open("file.txt" "r", encoding="utf8")`` (print is a bad example as the implicit str concat happens anyway).

Also seeing https://github.com/PyCQA/pylint/pull/1655#discussion_r140022976 I think not including calls is an oversight too.
